datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVDataMapProvider.scala

```diff
@@ -163,9 +165,47 @@ class MVDataMapProvider(
         SparkSQLUtil.execute(loadCommand, sparkSession)
       } catch {
         case ex: Exception =>
+          // If load to dataMap table fails, disable the dataMap and if newLoad is still
+          // in INSERT_IN_PROGRESS state, mark for delete the newLoad and update table status file
           DataMapStatusManager.disableDataMap(dataMapSchema.getDataMapName)
           LOGGER.error("Data Load failed for DataMap: ", ex)
-          return false
+          val segmentStatusManager = new SegmentStatusManager(dataMapTable
```

Comment:
Extract to a new method in `CarbonLoaderUtil`

Reply:
done